### PR TITLE
[4694] Add styles to the contact form

### PR DESCRIPTION
**Pull request by jvega190**
https://github.com/craftercms/craftercms/issues/4694

### CODE REVIEW

**Comment by jvega190** [comment]
@rart this is ready, but I can't assign you as reviewer

File: authoring/content-types/component/contact-form/form-definition.xml [hunk 1]
```diff
@@ -26,455 +26,524 @@
...
+								<type>string</type>
+							</constraint>
+						</constraints>
+					</field>
+					<field>
+						<type>checkbox</type>
+						<id>required_b</id>
+						<iceId></iceId>
+						<title>Required</title>
+						<description></description>
```
Comment (rart) [inline authoring/content-types/component/contact-form/form-definition.xml:383]: Add description

File: authoring/content-types/component/contact-form/form-definition.xml [hunk 2]
```diff
@@ -26,455 +26,524 @@
...
+					<constraint>
+						<name>pattern</name>
+						<value><![CDATA[]]></value>
+						<type>string</type>
+					</constraint>
+				</constraints>
+			</field>
+			<field>
+				<type>checkbox</type>
+				<id>usePlugintheme_b</id>
```
Comment (rart) [inline authoring/content-types/component/contact-form/form-definition.xml:501]: Fix case
```suggestion
				<id>usePluginTheme_b</id>

```

File: authoring/content-types/component/contact-form/form-definition.xml [hunk 3]
```diff
@@ -26,455 +26,524 @@
...
+				<iceId></iceId>
+				<title>Use Plugin theme</title>
```
Comment (rart) [inline authoring/content-types/component/contact-form/form-definition.xml:503]:
```suggestion
				<title>Use Plugin Theme</title>

```

File: authoring/content-types/component/contact-form/form-definition.xml [hunk 4]
```diff
@@ -26,455 +26,524 @@
...
+				<description>Include styles bundled with the plugin.</description>
```
Comment (rart) [inline authoring/content-types/component/contact-form/form-definition.xml:504]:
```suggestion
				<description>Include/exclude styles bundled with the plugin.</description>

```

File: authoring/content-types/component/contact-form/form-definition.xml [hunk 5]
```diff
@@ -26,455 +26,524 @@
...
+				<defaultValue>true</defaultValue>
+				<help>Disable selection if you have Bootstrap included on your site, or if you want to customize the plugin with your own styles.</help>
```
Comment (rart) [inline authoring/content-types/component/contact-form/form-definition.xml:506]:
```suggestion
				<help>Disable if your site styles already have the styles you wish to use for the plugin form (e.g. if your site uses Bootstrap, you don't need additional styles).</help>

```

File: readme.md [hunk 6]
```diff
@@ -43,4 +43,31 @@ The default Groovy controller included in this plugin will lookup a bean from th
 
 # Styling
 
-TBD
+The styles of the Contact Form plugin are based on Bootstrap 5 forms components, using Bootstrap's 
+[build tools](https://getbootstrap.com/docs/5.0/getting-started/build-tools/#sass):
+
+```
```
Comment (rart) [inline readme.md:49]:
```suggestion
```scss
```

File: delivery/templates/contact-form.ftl [hunk 7]
```diff
@@ -1,16 +1,44 @@
...
+        <label for="${contentModel.formId_s}-message" class="form-label">${contentModel.messageLabel_s}</label>
+        <textarea id="${contentModel.formId_s}-message" name="message" class="form-control" required></textarea>
+    </div>
+
+    <input type="hidden" name="formId" value="${contentModel.formId_s}"/>
+    <input type="submit" class="submit-btn" value="${contentModel.submitLabel_s}"/>
 </form>
+
+<#if validateForm>
+    <script src="/static-assets/plugins/org/craftercms/plugin/contactform/js/validate-form.js"></script>
```
Comment (rart) [inline delivery/templates/contact-form.ftl:43]: @jdrossl is this the right location for javascripts according to plugin design?

File: delivery/templates/contact-form.ftl [hunk 8]
```diff
@@ -1,16 +1,44 @@
...
+<#assign validateForm = contentModel.usePluginScript_b?? && contentModel.usePluginScript_b />
 
-  <label for="${contentModel.formId_s}-message">${contentModel.messageLabel_s}</label><br/>
-  <textarea id="${contentModel.formId_s}-message" name="message"></textarea><br/>
+<form
+    id="${contentModel.formId_s}"
+    action="${contentModel.url_s}"
+    method="post"
+    class="contact-form <#if validateForm>needs-validation</#if>"
+    <#if validateForm>novalidate</#if>
```
Comment (rart) [inline delivery/templates/contact-form.ftl:12]: Is this condition correct? You want novalidate if the model states to validate?

File: authoring/content-types/component/contact-form/form-definition.xml [hunk 9]
```diff
@@ -26,455 +26,524 @@
...
+						<type>boolean</type>
+					</constraint>
+				</constraints>
+			</field>
+			<field>
+				<type>checkbox</type>
+				<id>usePluginScript_b</id>
+				<iceId></iceId>
+				<title>Use Plugin Script</title>
+				<description>Enable form validation.</description>
```
Comment (rart) [inline authoring/content-types/component/contact-form/form-definition.xml:527]:
```suggestion
				<description>Enable required field validation via bundled plugin script.</description>

```

File: authoring/content-types/component/contact-form/form-definition.xml [hunk 10]
```diff
@@ -26,455 +26,524 @@
...
+				<defaultValue>true</defaultValue>
+				<help>Disable selection if you want to validate the form on your own.</help>
```
Comment (rart) [inline authoring/content-types/component/contact-form/form-definition.xml:529]:
```suggestion
				<help>Disable if you want to validate the form on your own or remove the validation.</help>

```

**Review by rart — CHANGES_REQUESTED** [review]

File: delivery/templates/contact-form.ftl (hunk 8 above)
Comment (jvega190) [reply delivery/templates/contact-form.ftl:12]: yeah, that's a prop that needs to be added to disable HTML5 default validation, to allow custom validation.

**Review by rart — APPROVED** [review]

**Pull request merged**